manifests/params.pp

```diff
@@ -2,7 +2,7 @@
 # @api private
 class puppetserver_foreman::params {
   $lower_fqdn = downcase($facts['networking']['fqdn'])
-  $foreman_url = "https://${lower_fqdn}"
+  $foreman_url = lookup('foreman_proxy::foreman_base_url') |$key| { "https://${lower_fqdn}" }
```

Comment:
this is just an odd way to write `lookup('foreman_proxy::foreman_base_url', {'default_value': "https://${lower_fqdn}"})`, right?

(according to https://www.puppet.com/docs/puppet/7/hiera_automatic.html#puppet_lookup -- just weird to read it that way)

Reply:
Yes, mostly because I didn't know you could do it that way. In this case it doesn't matter, but there certainly are cases for a lambda (like an expensive function call).

Reply:
oh yeah absolutely. just making sure I read things right :D

Reply:
And I also learned a think about `lookup()`. I'll admit that I just went through the docs, saw an example that did what I needed and didn't look further at the options.